libs/tokens/src/state/tokens/allTokensAtom.ts

```diff
@@ -0,0 +1,94 @@
+import { atom } from 'jotai'
+import { environmentAtom } from '../environmentAtom'
+import { TokensMap } from '../../types'
+import { NATIVE_CURRENCY_BUY_TOKEN, TokenWithLogo } from '@cowprotocol/common-const'
+import { tokenMapToList } from '../../utils/tokenMapToList'
+import { userAddedTokensAtom } from './userAddedTokensAtom'
+import { favouriteTokensAtom } from './favouriteTokensAtom'
+import { listsEnabledStateAtom, listsStatesListAtom } from '../tokenLists/tokenListsStateAtom'
+
+export interface TokensByAddress {
+  [address: string]: TokenWithLogo
+}
+
+export interface TokensBySymbol {
+  [address: string]: TokenWithLogo[]
+}
+
+export interface TokensState {
+  activeTokens: TokensMap
+  inactiveTokens: TokensMap
+}
+
+export const tokensStateAtom = atom<TokensState>((get) => {
+  const { chainId } = get(environmentAtom)
+  const listsStatesList = get(listsStatesListAtom)
+  const listsEnabledState = get(listsEnabledStateAtom)
+
+  return listsStatesList.reduce<TokensState>(
+    (acc, list) => {
+      const isListEnabled = listsEnabledState[list.id]
+
+      list.list.tokens.forEach((token) => {
+        if (token.chainId !== chainId) return
+
+        const tokenAddress = token.address.toLowerCase()
+
+        if (isListEnabled) {
+          acc.activeTokens[tokenAddress] = token
+        } else {
+          acc.inactiveTokens[tokenAddress] = token
+        }
+      })
+
+      return acc
+    },
+    { activeTokens: {}, inactiveTokens: {} }
+  )
+})
+
+export const activeTokensAtom = atom<TokenWithLogo[]>((get) => {
```

Comment:
It would be desireable a small phrase as JSdoc on what this derived atom return exactly: like an alphabetically ordered list of tokens with images, concatenating the the active + user added + favourite tokens with the native to as the first item.

Reply:
Added: https://github.com/cowprotocol/cowswap/pull/3201/commits/20af0e7a3782ed1c30b680532a3f68168375b495